File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/DeviceEventSample/src/com/hp/workpath/sample/deviceeventsample/fragment/ResponseInterface.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.deviceeventsample.fragment;

import com.hp.workpath.api.Result;
import com.hp.workpath.api.device.events.DeviceEvent;

import java.util.List;

public interface ResponseInterface {
    void success(List<DeviceEvent> deviceEvents);
    void failure(String msg, Result result);
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/DeviceEventSample/src/com/hp/workpath/sample/deviceeventsample/fragment/ResponseInterface.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.deviceeventsample.fragment

import com.hp.workpath.api.Result
import com.hp.workpath.api.device.events.DeviceEvent

interface ResponseInterface {
    fun success(deviceEvents: List<DeviceEvent>?)
    fun failure(msg: String?, result: Result?)
}